subworkflows/local/prepare_input_stitch/prepare_input_stitch.nf

```diff
@@ -0,0 +1,64 @@
+include { BCFTOOLS_QUERY                    } from '../../../modules/nf-core/bcftools/query/main'
+include { BCFTOOLS_NORM                     } from '../../../modules/nf-core/bcftools/norm/main'
+include { GAWK                              } from '../../../modules/nf-core/gawk'
+
+
+workflow PREPARE_INPUT_STITCH {
+
+    take:
+    ch_panel_sites
+    ch_fasta
+    ch_input_impute
+
+    main:
+
+    ch_versions      = Channel.empty()
+
+    // Prepare posfile and parameters for STITCH
+    // Convert position file to tab-separated file
+    BCFTOOLS_QUERY(ch_panel_sites, [], [], [])
+    ch_posfile = BCFTOOLS_QUERY.out.output
+
+    // Remove multiallelic positions from tsv
+    GAWK(ch_posfile, [])
+
+    // Get chromosomes of posfile
+    ch_posfile = GAWK.out.output.map{meta, posfile -> return[['chr': meta.chr], posfile]}
+
+    // Get chromosomes of fasta
+    ch_chromosomes = ch_fasta.map{it -> it[2]}
```

Comment:
If I'm not wrong this chromosome extraction is done on the initialisation level.
We could add a channel as input to link it ?

Reply:
Good idea, I'll check this!